app/models/meursing_lookup/steps/base.rb

```diff
@@ -0,0 +1,66 @@
+module MeursingLookup
+  module Steps
+    class Base < WizardSteps::Step
+      def self.inherited(child_class)
+        child_class.attribute child_class.key, :string
+        child_class.validates child_class.key, presence: true
+
+        super
+      end
+
+      def options
```

Comment:
This method is doing several things and is quite hard to follow. Would it make sense to separate some of the concerns - so move the tree to its own class and add some methods that class for querying the dataset - that may help to simplify the logic here. 

```
class MeursingCodeTree
  def self.load_default
      new(File.read(....))
  end
end

module TradeTariffBackend
  def meursing_code_tree
    @meursing_code_tree ||= MeursingCodeTree.load_default
  end
end
```

Reply:
Part of me wonders if this code is generic but without needing to be - if there's a generic 'query' option on a `MeursingCodeTree` - would that mean each of the steps could have their own `def options` passing in relevant params to the `MeursingCodeTree#lookup` ?